### ISSUE TITLE
Clear Geometry implementation

### ISSUE THREAD

**Issue by JulReinhardt** [lightsources/NX-cxi-ptycho#14]

**Comment by JulReinhardt** [comment]
@prjemian Is an axis a group or a dataset?
I assume each axis should be a dataset of the NXtransformation group. Like here: https://manual.nexusformat.org/design.html#coordinate-transformation-field-and-attributes However, it differs a bit from the usual structure, where you have the group and then a name for the dataset e.g. distance plus the matching value e.g. 1 plus the attribute unit "m"

There are no values for the axis dataset as it uses attributes to hold the vectors etc.
So how would we create the dataset "axis" then? Would we use name for the dataset is "axis_name_1" with the e.g.  value "x_translation" and then the attributes "vector", "offset" etc? 
Or would we create an anonymous dataset i.e. not filling the name argument and only using unambiguous values e.g. x_translation" etc.?

See h5py group class method
```
    def create_dataset(self, name, shape=None, dtype=None, data=None, **kwds):
        """ Create a new HDF5 dataset

        name
            Name of the dataset (absolute or relative).  Provide None to make
            an anonymous dataset.
```

**Comment by carterbox** [comment]
Yes, NXtransformation is a group and each axis in the group is a dataset (or a link to a dataset).

The value of the axis is the magnitude of movement along the axis. The `vector` attribute is a unit vector; it is direction information only. The axis dataset cannot be empty because otherwise, the description of the transformation is incomplete because it would not have a magnitude.

**Comment by JulReinhardt** [comment]
Ok, the what is the offset supposed to represent? I understood the offset to be the magnitude of movement, although I am happy to use the value instead.

**Comment by prjemian** [comment]
Offset is translation. If no translation (only rotation), then set to 0,0,0

**Comment by carterbox** [comment]
Offsets used because stages are of finite size. Depending on how you stack the physical stages together, you may need a fixed-offset to get the rotation centers and zero-positions aligned correctly.
